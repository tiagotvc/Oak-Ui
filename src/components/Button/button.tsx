import React from "react";
import { ButtonProps } from "./button.utils";
import { BtnComponent } from "./button.styles";

export const Button: React.FC<ButtonProps> = ({ children, type }, props) => {
  return (
    <BtnComponent type={type} {...props}>
      {children}
    </BtnComponent>
  );
};

export default Button;
