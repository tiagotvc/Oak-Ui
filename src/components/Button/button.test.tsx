import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import { Button } from "./button";
import renderer from "react-test-renderer";
import "jest-styled-components";

describe("Componente Btn all tests", () => {
  it("Should render a react Button componente as primary type", () => {
    const onClickFn = jest.fn();
    const eventMock = { preventDefault: jest.fn() };

    //Render tests
    render(<Button type="primary">Primary</Button>);
    const primaryBtn = screen.getByText(/Primary/i);
    expect(primaryBtn).toBeVisible();
    expect(primaryBtn).toHaveAttribute("type", "primary");
    expect(primaryBtn).toBeInTheDocument();

    //Styles tests
    const tree = renderer
      .create(
        <Button type="primary" test-id="1">
          Primary
        </Button>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("background", "#00bfff");
    expect(tree).toHaveStyleRule("opacity", "85%", { modifier: ":hover" });

    //Functional tests
    const clickedBtn = renderer.create(
      <Button type="primary" test-id="1" onClick={onClickFn}>
        Primary
      </Button>
    ).root;
    clickedBtn.props.onClick(eventMock);
    fireEvent.click(primaryBtn);
    expect(onClickFn).toBeCalled();
  });
});
