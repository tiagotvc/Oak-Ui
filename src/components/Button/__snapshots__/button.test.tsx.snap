// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Btn all tests Should render a react Button componente as primary type 1`] = `
.c0 {
  display: flex;
  width: 140px;
  height: 35px;
  justify-content: center;
  align-items: center;
  font-size: 15px;
  font-weight: bold;
  color: white;
  border-radius: 10px;
  border: none;
  cursor: pointer;
  background: #00bfff;
}

.c0:hover {
  opacity: 85%;
  border: 1px solid lightgray;
}

<button
  className="c0"
  type="primary"
>
  Primary
</button>
`;
